### PR TITLE
Don't close session

### PR DESCRIPTION
**Pull request by dominikl**
Don't close session if it wasn't initialized by gateway.

See https://trello.com/c/Dbl1c1LQ/543-java-gateway-disconnect-no-close and [integration test](https://github.com/openmicroscopy/openmicroscopy/pull/6059).

### CODE REVIEW

File: src/main/java/omero/gateway/Connector.java [hunk 1]
```diff
@@ -681,8 +681,9 @@ void joinSession()
      * 
      * @param networkup Pass <code>true</code> if the network is up,
      * <code>false</code> otherwise.
+     * @param closeSession Pass <code>true</code> to close the session
      */
-     void close(boolean networkup)
```
Comment (joshmoore) [inline src/main/java/omero/gateway/Connector.java]: For this not to be a breaking change, you'll need to keep the old method signature and call the new method with a default (optionally adding a `Deprecated` annotation)

Comment (dominikl) [reply src/main/java/omero/gateway/Connector.java]: Can do. But it's not a public method, the Connector class isn't really meant to be used directly, only to be used by the Gateway.

Comment (joshmoore) [reply src/main/java/omero/gateway/Connector.java]: Hmm... ok. Fair enough.

Comment (dominikl) [reply src/main/java/omero/gateway/Connector.java]: But I'll add the previous method as deprecated again, doesn't do any harm.

**Comment by joshmoore** [comment]
The remote calls make sense :+1:

Reading through the comments, I wondered if having the `isSessionLogin` state on the `Connector` would simplify some of your code. For example, you can't call `closeSessionOnExit` until the connector exists and you're passing isSessionLogin _into_ the Connector in most locations.

**Comment by dominikl** [comment]
That's true. I'll move it into the Connector.

File: src/main/java/omero/gateway/Gateway.java [hunk 2]
```diff
@@ -1701,4 +1724,31 @@ public void close() throws Exception {
         if (connected)
             disconnect();
     }
+
+    /**
+     * Wrapper holding a client and a flag if the client is hooked up
+     * to a already existing session.
+     */
+    private class SessionWrapper {
```
Comment (joshmoore) [inline src/main/java/omero/gateway/Gateway.java]: Interesting. Does this actually mean you'd benefit from having the `sessionLogin` flag in `omero.client` itself?

Comment (dominikl) [reply src/main/java/omero/gateway/Gateway.java]: Well... yes, in the end it actually could be a useful property of the `omero.client` itself.

**Comment by dominikl** [comment]
I still think this PR makes sense, until this functionionality gets moved into the `client` itself.

**Comment by joshmoore** [comment]
:+1: Merging for 5.5.3. Plan is to get integration tests green before releasing.

**Pull request merged**